newIDE/app/src/UI/ImagesDisplay.js

```diff
@@ -0,0 +1,255 @@
+  )();
+
+  const [
+    mobileGridClientWidth,
+    setMobileGridClientWidth,
+  ] = React.useState<number>(0);
+  const [mobileGridScrollX, setMobileGridScrollX] = React.useState(0);
+  const [
+    currentlyViewedImageIndex,
+    setCurrentlyViewedImageIndex,
+  ] = React.useState<number>(0);
+
+  const mobileImageWidth =
+    mobileGridClientWidth -
+    30 - // Width kept for user to see that there's an image after or before
+    (horizontalOuterMarginToEatOnMobile || 0);
+
+  React.useEffect(
+    () => {
+      setCurrentlyViewedImageIndex(
+        Math.round(mobileGridScrollX / (mobileImageWidth + GRID_SPACING))
+      );
+    },
+    [mobileImageWidth, mobileGridScrollX]
+  );
+
+  if (isMobile) {
+    return (
+      <div
+        style={{
+          ...styles.flexColumn,
+          marginLeft: horizontalOuterMarginToEatOnMobile
+            ? -horizontalOuterMarginToEatOnMobile
+            : 0,
+          marginRight: horizontalOuterMarginToEatOnMobile
+            ? -horizontalOuterMarginToEatOnMobile
+            : 0,
+        }}
+      >
+        <Measure
+          bounds
+          onResize={contentRect => {
+            setMobileGridClientWidth(contentRect.bounds.width);
+          }}
+        >
+          {({ contentRect, measureRef }) => (
+            <div style={styles.flex} ref={measureRef}>
+              <Grid
+                classes={classesForGridContainer}
+                container
+                spacing={GRID_SPACING}
+                wrap="nowrap"
+                style={styles.mobileGrid}
+                onScroll={(event: SyntheticEvent<HTMLDivElement>) =>
+                  setMobileGridScrollX(event.currentTarget.scrollLeft)
+                }
+              >
+                {imagesUrls.map((url, index) => (
+                  <Grid
+                    item
+                    key={url}
+                    style={{
+                      scrollSnapAlign: horizontalOuterMarginToEatOnMobile
+                        ? 'center'
+                        : 'start',
+                    }}
+                  >
+                    <CardMedia>
+                      <CorsAwareImage
+                        src={url}
+                        style={{
+                          ...styles.mobileImageCarouselItem,
+                          height: mobileImageWidth / (16 / 9),
+                        }}
+                        alt={altTextTemplate.replace(
+                          /{imageIndex}/g,
+                          String(selectedImageIndex + 1)
+                        )}
+                      />
+                    </CardMedia>
+                  </Grid>
+                ))}
+              </Grid>
+            </div>
+          )}
+        </Measure>
+        <Line justifyContent="center">
+          <Text noMargin size="body2">
+            {currentlyViewedImageIndex + 1}/{imagesUrls.length}
+          </Text>
+        </Line>
+      </div>
+    );
+  }
+  return (
+    <Column noMargin>
```

Comment:
nit: rather than adding a line around the Grid, you can use a ColumnStackLayout which inserts spacer between child (and no spacer around them, so the component does not introduce unnecessary paddings that make things hard to align).

Reply:
Oh good idea! Thanks